expands readme to expand what GSP is and isn't (#249)
alphagov/gsp: README.md

# GSP [![IRC](https://img.shields.io/badge/kubernetes-v1.12.6-0099ef.svg)]() <img align="right" src="./docs/assets/gsp.png" alt="gsp" width="200" height="whatever">

> GSP is a Kubernetes distribution maintained by the [Govenement Digital Service](https://www.gov.uk/government/organisations/government-digital-service) created to meet the common needs of running digital services in production.

Reviewer: ```suggestion
The GDS Supported Platform (GSP) is a Kubernetes distribution maintained by the [Govenement Digital Service](https://www.gov.uk/government/organisations/government-digital-service) created to meet the common needs of running digital services in production.
```

Author: Damn was hoping nobody had noticed I stopped calling it that hehe... Are we _seriously_ sticking with an initialism that has another initialism in it heh ... I won't start a naming bikeshed here and I will conceed to mention it's backryomned name as you suguest ... But I do think something less abrasive to PaaS is required (they are a gds supported platform too!)

Author: expanded :done:

> GSP (GDS Supported Platform) is a Kubernetes distribution maintained by the [Govenement Digital Service](https://www.gov.uk/government/organisations/government-digital-service) created to meet the common needs of running digital services in production.

Reviewer 2: typo "Govenement Digital Service" should read "Government Digital Service"

> GSP (GDS Supported Platform) is a Kubernetes distribution maintained by the [Govenement Digital Service](https://www.gov.uk/government/organisations/government-digital-service) created to meet the common needs of running digital services in production.

Reviewer 2: @chrisfarms I think we need to be more explicit about who this is for. i.e. GDS Service Teams

Author: Our needs have def come from GDS services... but it _could_ be used by others who want to follow the same path 🤷‍♂

* Reduce onboarding/support burden by sharing a consistent base and common declarative language
* Reduce costs to programmes by sharing infrastructure where possible
* Minimise vendor lock-in while leveraging managed services by using non-proprietary configuration language and abstractions
* Improve service team delivery by providing release automation and observability tooling

> * Increase service team confidence in deployments by enabling strong parity between local developement and production environments

Reviewer: ```suggestion
* Increase service team confidence in deployments by enabling strong parity between local development and production environments
```

GSP provides a suite of pre-configured components along with upstream Kubernetes, much like a GNU/Linux distribution provides a suite of userspace components along with the upstream Linux kernel.

## Features

- A declarative continuous delivery workflow - merging to master triggers deployment to production
- A container platform based on industry standard [Docker](https://docs.docker.com/) and [Kubernetes](https://kubernetes.io)
- Build and release automation powered by [ConcourseCI](https://concourse-ci.org/)
- A private container registry with [Docker Registry](https://docs.docker.com/registry/)
- Signing of docker image integrity with [Docker Notary](https://docs.docker.com/notary/)
- Scanning of docker images for security vulnerabilities with [clair](https://github.com/coreos/clair)
- Monitoring and alerting with [Prometheus](https://prometheus.io/), [Alertmanager](https://prometheus.io/docs/alerting/alertmanager/) and [Grafana](https://grafana.com/)
- Secure git-based secrets configuration with [sealed-secrets](https://github.com/bitnami-labs/sealed-secrets)

> - Ingress management with [nginx ingress controller](https://kubernetes.github.io/ingress-nginx/)

Reviewer 2: remove the nginx ingress line and replace with

- Ingress management and service mesh with [istio](https://istio.io/)

- Protective monitoring provided by GDS TechOps CyberSecurity with [Splunk](https://www.splunk.com/)

> - Cloud infrastructure hosted on [AWS](https://aws.amazom.com) across multipleavailability zones in the London

Reviewer: ```suggestion
- Cloud infrastructure hosted on [AWS](https://aws.amazom.com) across multiple availability zones in London
```

> - Kubernetes control plane with [AWS EKS](https://aws.amazon.com/eks/)

Reviewer 2: add a line to mention that it uses the AWS Global Accellerator

- [AWS Global Accellerator](https://aws.amazon.com/global-accelerator/) for static IP routing

Author: The GA isn't really part of the platform imo ... I think it's just something you bolt on in the account if you want it ... it's probably overkill unless you are doing a replatforming or intend to do some kind of multi-cloud madness

## Non goals

GSP is not a managed service.

> If you are a team looking for a fully managed platform, we recormend you evaluate [GOV.UK PaaS](https://docs.cloud.service.gov.uk) before attempting to run and manage your own GSP instance. 

Reviewer: ```suggestion
If you are a team looking for a fully managed platform, we recommend you evaluate [GOV.UK PaaS](https://docs.cloud.service.gov.uk) before attempting to run and manage your own GSP instance. 
```

The platform has been designed to complement an organisation that practices a Reliability Eningeering model that assumes there exists a small number of infrastructure and reliability focused members capable of supporting a much larger team or programme.

![](./docs/assets/paas-spectrum.png)

The figure above illustrates where we think GSP fits on a "PaaS Spectrum":

> * On the right-hand-side we have the situation where service teams all design their deployment architectures in isolation using (hopefully) cloud managed services which gives the ultimate in flexability at the cost of poor knowledge share accross the organisation and a need for dedicated infrastructure expertise.

Reviewer: ```suggestion
* On the right-hand-side we have the situation where service teams all design their deployment architectures in isolation using (hopefully) cloud managed services which gives the ultimate in flexibility at the cost of poor knowledge share accross the organisation and a need for dedicated infrastructure expertise.
```

> * On the right-hand-side we have the situation where service teams all design their deployment architectures in isolation using (hopefully) cloud managed services which gives the ultimate in flexibility at the cost of poor knowledge share accross the organisation and a need for dedicated infrastructure expertise.

Reviewer 2: typo  "accross" -> "across"

> * On the left-hand-side we have the fully managed GOV.UK PaaS platform where a service team may not need any infrastructure expertise but at the cost of flexability.

Reviewer 2: typo "flexability" -> "flexibility"

* Sitting between the two is GSP which is:

> 	* more complex for service teams compared to PaaS, in exchange for greater flexability over the platform itself

Reviewer 2: typo "flexability" -> "flexibility"

	* more opinionated than a bespoke architecture, in exchange for greater knowledge/code sharing between teams
	* more isolated than GOV.UK PaaS, but encourages sharing as much as possible 

## Getting started

> - [Host a GDS Supported Platform cluster in the AWS cloud](docs/gds-supported-platform/getting-started-gsp-cluster.md)

Reviewer 2: Can we change the title of this link so that its consistent with the content it links to

- [Getting started with a GDS Supported Platform cluster](docs/gds-supported-platform/getting-started-gsp-cluster.md)

> - [Host a GDS Supported Platform cluster locally](docs/gds-supported-platform/getting-started-gsp-local.md)

Reviewer 2: similar to the link about, I propose we  reuse the actual title of the content that it links to

- [Getting started with a local GDS Supported Platform development cluster](docs/gds-supported-platform/getting-started-gsp-local.md)
